Source/WebCore/Modules/webaudio/MediaStreamAudioSourceGStreamer.cpp

```diff
@@ -58,22 +58,24 @@ void MediaStreamAudioSource::consumeAudio(AudioBus& bus, size_t numberOfFrames)
     MediaTime mediaTime((m_numberOfFrames * G_USEC_PER_SEC) / m_currentSettings.sampleRate(), G_USEC_PER_SEC);
     m_numberOfFrames += numberOfFrames;
 
-    GstAudioInfo info;
-    gst_audio_info_set_format(&info, GST_AUDIO_FORMAT_F32LE, m_currentSettings.sampleRate(), bus.numberOfChannels(), nullptr);
-    GST_AUDIO_INFO_LAYOUT(&info) = GST_AUDIO_LAYOUT_NON_INTERLEAVED;
-    size_t size = GST_AUDIO_INFO_BPS(&info) * bus.numberOfChannels() * numberOfFrames;
+    // Lazily initialize caps, the settings don't change so this is OK.
+    if (!m_caps || GST_AUDIO_INFO_CHANNELS(&m_info) != static_cast<int>(bus.numberOfChannels())) {
```

Comment:
Perhaps we need a `gst_audio_info_init(&info)` in the constructor? Else, `GST_AUDIO_INFO_CHANNELS(&m_info)` uninitialized in the first iteration.

Reply:
Yes, but since we test caps first I think we're OK? if caps is null, then m_info as well.